lib/cli.js

```diff
@@ -127,17 +129,31 @@ function buildWatcherOptions(options) {
   if (!options) {
     options = {};
   }
-  const watcher = options.watcher;
+  let watcher = options.watcher;
 
   if (watcher && ['polling', 'watchman', 'node', 'events'].indexOf(watcher) === -1) {
     throw new Error(`Unknown watcher type --watcher=[polling|watchman|node] but was: ${watcher}`);
   }
 
+  if (!watcher) {
+    try {
+      const output = childProcess.execSync('watchman version');
+      const version = JSON.parse(output).version;
+      if (!semver.satisfies(version, '>= 3.7.0')) {
+        throw new Error('watchman version older than 3.7.0');
```

Comment:
When auto-detecting, it may not be a good idea to error. Rather it should be considered when deciding `node` vs `watchman`. The reason is, other executables can be named `watchman`... there is exists one of these on the npm registry...

I think this will all be correctly handled once we extract watch-detector.